packages/react/src/useTable.tsx

```diff
@@ -248,10 +149,105 @@ const getRows = (props: { data: GadgetRecord<any>[]; columns: TableColumn[] }) =
   const { columns, data } = props;
 
   return data.map((record) => {
-    const row: Record<string, ColumnValueType> = { id: record.id };
+    const row: TableRow = { id: record.id };
     for (const { apiIdentifier, getValue } of columns) {
       row[apiIdentifier] = getValue(record);
     }
     return row;
   });
 };
+
+const getTableColumnArray = (fields: FieldMetadataFragment[], columnsFromOptions: TableOptions["columns"]) => {
```

Comment:
this should be in a useMemo or useCallback or it will be regenerated on every re-render

Reply:
I also notice this is true of `getTableData` and `getRows`.

Reply:
`getTableColumnArray` is memoised with `useMemo` when calling the function.

Reply:
That doesn't really matter, it's the definition of the function that needs to be memoized or it will be re-allocated every render cycle.

Reply:
or the function can be moved outside of the hook scope if it doesn't use any of the state other than it's params, which I would say it probably shouldn't

Reply:
disregard this, misread the closure scope, it's already outside the hook.